server/src/main/java/org/opentosca/toscana/plugins/cloudformation/mapper/CapabilityMapper.java

```diff
@@ -0,0 +1,310 @@
+            if (osCapability.getDistribution().get().equals(OsCapability.Distribution.UBUNTU)) {
+                // */ubuntu/images/* gets better results than plain *ubuntu*
+                describeImagesRequest.withFilters(new Filter("name").withValues("*ubuntu/images/*"));
+            } else {
+                //just search for the string
+                describeImagesRequest.withFilters(new Filter("name").withValues("*" + osCapability.getDistribution()
+                    .toString() + "*"));
+            }
+        }
+        if (osCapability.getVersion().isPresent()) {
+            describeImagesRequest.withFilters(new Filter("name").withValues("*" + osCapability.getVersion().get() +
+                "*"));
+        }
+        if (osCapability.getArchitecture().isPresent()) {
+            if (osCapability.getArchitecture().get().equals(OsCapability.Architecture.x86_64)) {
+                describeImagesRequest.withFilters(new Filter("architecture").withValues("x86_64"));
+            } else if (osCapability.getArchitecture().get().equals(OsCapability.Architecture.x86_32)) {
+                describeImagesRequest.withFilters(new Filter("architecture").withValues("i386"));
+            }
+        } else {
+            //defaulting to 64 bit architecture
+            describeImagesRequest.withFilters(new Filter("architecture").withValues("x86_64"));
+        }
+        String imageId;
+        try {
+            DescribeImagesResult describeImagesResult = ec2.describeImages(describeImagesRequest);
+            Integer numReceivedImages = describeImagesResult.getImages().size();
+            logger.debug("Got " + numReceivedImages + " images from aws");
+            if (numReceivedImages > 0) {
+                DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+                Map<Date, Image> creationDateMap = new HashMap<>();
+                for (Image image : describeImagesResult.getImages()) {
+                    try {
+                        Date date = dateFormat.parse(image.getCreationDate());
+                        creationDateMap.put(date, image);
+                    } catch (ParseException pE) {
+                        logger.error("Error parsing dateformat");
+                        pE.printStackTrace();
+                    }
+                }
+                Image latest = creationDateMap.get(Collections.max(creationDateMap.keySet()));
+                logger.debug("Latest image received: " + latest.toString());
+                imageId = latest.getImageId();
+            } else {
+                logger.warn("No images received defaulting to old ubuntu 16.04 image");
+                imageId = "ami-0def3275";
+                //TODO maybe not defaulting but throwing a transformation failed exception?
+            }
+        } catch (SdkClientException se) {
+            logger.error("Cannot connect to AWS to request image Ids, defaulting to old ubuntu 16.04 image");
```

Comment:
Is there a problem if the user want a Windows system and you set the default to a ubuntu version?